k9mail/src/main/java/com/fsck/k9/activity/setup/outgoing/AccountSetupOutgoing.java

```diff
@@ -437,47 +375,35 @@ private void updateAuthPlainTextFromSecurityType(ConnectionSecurity securityType
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (resultCode == RESULT_OK) {
```

Comment:
That depends. Usually, the flow will be something like `MvpView.showSomeDialog()` which will lead to a callback to `Presenter.onSomeDialogDismissed()` or similar. The view definitely shouldn't decide what to do (like the `finish()` here)

Sometimes it's easier to just hand through the `onActivityResult`, but that leads to a stronger coupling between the android view code and presenter, and should not be the typical case.